pkg/trace/sampler/dynamic_config.go

```diff
@@ -24,45 +28,90 @@ func NewDynamicConfig() *DynamicConfig {
 	return &DynamicConfig{RateByService: RateByService{}}
 }
 
+// State specifies the current state of DynamicConfig
+type State struct {
+	Rates      map[string]float64
+	Mechanisms map[string]uint32
```

Comment:
How can someone figure out what a mechanism is? Does it need its own type?

Reply:
Yeah, let's create a type for it.

Reply:
Hey, you resolved the comment, but there are multiple places where you are still using `uint32` instead of `samplingMechanism`